src/bin/grin.rs

```diff
@@ -372,7 +372,7 @@ fn wallet_command(wallet_args: &ArgMatches) {
 			let mut contents = String::new();
 			file.read_to_string(&mut contents)
 				.expect("Unable to read transaction file.");
-			wallet::receive_json_tx(&wallet_config, &keychain, contents.as_str()).unwrap();
+			wallet::receive_json_tx_str(&wallet_config, &keychain, contents.as_str()).unwrap();
```

Comment:
Split `receive_json_tx` out into `receive_json_tx` and `receive_json_tx_str`.
One takes a `JSONPartialTx` and one take json string and first converts it to `JSONPartialTx`.
We need the string version for wallet send/receive via a file.
The other works cleaner (and simpler) via the api (wth parsing handled via bodyparser).